### DISCUSSION TITLE
Error in loading "graph-compress.pb" when using dipole tensors

### DISCUSSION THREAD

**Discussion by yunai2384** [deepmodeling/deepmd-kit#2327] [Q&A] [answered]
I run the example of dipole in "**deepmd-kit-2.1.5/examples/water_tensor/dipole/dipole_input.json**", then the error occurred when loading the compressed or uncompressed graph.pb in LAMMPS by using **deepmd-kit-2.1.5/bin/lmp**. 

****use deepmd-kit at: Anaconda/envs/deepmd2.1.5ERROR: DeePMD-kit Error: TensorFlow Error: INVALID_ARGUMENT: Tensor fitting_attr/dfparam:0, specified in either feed_devices or fetch_devices was not found in the Graph (/home/conda/feedstock_root/build_artifacts/libdeepmd_1663923130507/work/source/lmp/pair_deepmd.cpp:840)**

I installed deepmd-kit 2.1.5 from conda.
How can I solve it? thanks.

### Command sequence:
        $ conda create -n deepmd2.1.5 deepmd-kit=2.1.5=*cpu libdeepmd=2.1.5=*cpu lammps horovod -c https://conda.deepmodeling.com -c defaults
        $ conda activate deepmd2.1.5
        $ cd deepmd-kit-2.1.5/examples/water_tensor/dipole
        $ dp train dipole_input.json
        $ dp freeze -o graph.pb
        $ dp  compress -i graph.pb -o graph-compress.pb
        $ lmp -in in.lammps

### the input file  **in.lammps**:
        units           metal
        boundary        p p p
        atom_style      atomic
        neighbor        2.0 bin
        neigh_modify    every 10 delay 0 check no
        read_data       water.lmp
        mass            1 16
        mass            2 2
        pair_style      deepmd graph-compress.pb
        pair_coeff  * *

### The input file dipole_input.json
        {
	        "_comment": " model parameters",
	        "model": {
		        "type_map": [
			        "O",
			        "H"
		        ],
		        "descriptor": {
			        "type": "se_e2_a",
			        "sel": [
				        46,
				        92
			        ],
			        "rcut_smth": 3.80,
			        "rcut": 4.00,
			        "neuron": [
				        25,
				        50,
				        100
			        ],
			        "resnet_dt": false,
			        "axis_neuron": 6,
			        "seed": 1,
			        "_comment": " that's all"
		        },
		        "fitting_net": {
			        "type": "dipole",
			        "sel_type": [
				        0
			        ],
			        "neuron": [
				        100,
				        100,
				        100
			        ],
			        "resnet_dt": true,
			        "seed": 1,
			        "_comment": " that's all"
		        },
		        "_comment": " that's all"
	        },
	        "learning_rate": {
		        "type": "exp",
		        "start_lr": 0.01,
		        "decay_steps": 5000,
		        "_comment": "that's all"
	        },
	        "loss": {
		        "type":"tensor",
		        "pref":			1.0,
		        "pref_atomic":	1.0,
		        "_comment": " that's all"
	        },
	        "_comment": " traing controls",
	        "training": {
		        "training_data": {
			        "systems":		[
				        "./training_data/atomic_system",
				        "./training_data/global_system"],
			        "batch_size":	"auto",
			        "_comment":		"that's all"
		        },
		        "validation_data":{
			        "systems":		[
				        "./validation_data/atomic_system",
				        "./validation_data/global_system"],
			        "batch_size":	1,
			        "numb_btch":	3,
			        "_comment":		"that's all"
		        },
		        "numb_steps":	2000,
		        "seed":		10,
		        "disp_file":	"lcurve.out",
		        "disp_freq":	100,
		        "save_freq":	1000,
		        "_comment":	"that's all"
	        },
	        "_comment": "that's all"
        }

### The error in log.lammps:
        LAMMPS (23 Jun 2022 - Update 1)
        OMP_NUM_THREADS environment is not set. Defaulting to 1 thread. (src/comm.cpp:98)
          using 1 OpenMP thread(s) per MPI task
        Loaded 1 plugins from /home/yunai/E/PG/Anaconda/envs/deepmd2.1.5/lib/deepmd_lmp
        Loaded 0 plugins from
        # bulk water
        
        units           metal
        boundary        p p p
        atom_style      atomic
        
        neighbor        2.0 bin
        neigh_modify    every 10 delay 0 check no
        
        read_data       water.lmp
        Reading data file ...
          triclinic box = (0 0 0) to (12.4447 12.4447 12.4447) with tilt (0 0 0)
          1 by 1 by 1 MPI processor grid
          reading atoms ...
          192 atoms
          read_data CPU = 0.001 seconds
        mass            1 16
        mass            2 2
        pair_style      deepmd graph-compress.pb
        Summary of lammps deepmd module ...
          >>> Info of deepmd-kit:
          installed to:       /home/yunai/E/PG/Anaconda/envs/deepmd2.1.5
          source:             v2.1.5
          source branch:       HEAD
          source commit:      6e3d4a62
          source commit at:   2022-09-23 16:10:28 +0800
          surpport model ver.:1.1
          build float prec:   double
          build variant:      cpu
          build with tf inc:  /home/yunai/E/PG/Anaconda/envs/deepmd2.1.5/include;/home/yunai/E/PG/Anaconda/envs/deepmd2.1.5/include
          build with tf lib:  /home/yunai/E/PG/Anaconda/envs/deepmd2.1.5/lib/libtensorflow_cc.so;/home/yunai/E/PG/Anaconda/envs/deepmd2.1.5/lib/libtensorflow_framework.so
          set tf intra_op_parallelism_threads: 775364658
          set tf inter_op_parallelism_threads: 873085232
          >>> Info of lammps module:
          **use deepmd-kit at:  /home/yunai/E/PG/Anaconda/envs/deepmd2.1.5ERROR: DeePMD-kit Error: TensorFlow Error: INVALID_ARGUMENT: Tensor fitting_attr/dfparam:0, specified in either feed_devices or fetch_devices was not found in the Graph (/home/conda/feedstock_root/build_artifacts/libdeepmd_1663923130507/work/source/lmp/pair_deepmd.cpp:840)**
        Last command: pair_style        deepmd graph-compress.pb

### The training log:
        WARNING:tensorflow:From /home/yunai/E/PG/Anaconda/envs/deepmd2.1.5/lib/python3.10/site-packages/tensorflow/python/compat/v2_compat.py:107: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.
        Instructions for updating:
        non-resource variables are not supported in the long term
        WARNING:root:To get the best performance, it is recommended to adjust the number of threads by setting the environment variables OMP_NUM_THREADS, TF_INTRA_OP_PARALLELISM_THREADS, and TF_INTER_OP_PARALLELISM_THREADS.
        WARNING:root:Environment variable KMP_BLOCKTIME is empty. Use the default value 0
        WARNING:root:Environment variable KMP_AFFINITY is empty. Use the default value granularity=fine,verbose,compact,1,0
        /home/yunai/E/PG/Anaconda/envs/deepmd2.1.5/lib/python3.10/importlib/__init__.py:169: UserWarning: The NumPy module was reloaded (imported a second time). This can in some cases result in small but subtle issues and is discouraged.
          _bootstrap._exec(spec, module)
        DEEPMD INFO    Calculate neighbor statistics... (add --skip-neighbor-stat to skip this step)
        OMP: Info #155: KMP_AFFINITY: Initial OS proc set respected: 0-11
        OMP: Info #216: KMP_AFFINITY: decoding x2APIC ids.
        OMP: Info #157: KMP_AFFINITY: 12 available OS procs
        OMP: Info #158: KMP_AFFINITY: Uniform topology
        OMP: Info #287: KMP_AFFINITY: topology layer "LL cache" is equivalent to "socket".
        OMP: Info #192: KMP_AFFINITY: 1 socket x 6 cores/socket x 2 threads/core (6 total cores)
        OMP: Info #218: KMP_AFFINITY: OS proc to physical thread map:
        OMP: Info #172: KMP_AFFINITY: OS proc 0 maps to socket 0 core 0 thread 0 
        OMP: Info #172: KMP_AFFINITY: OS proc 1 maps to socket 0 core 0 thread 1 
        OMP: Info #172: KMP_AFFINITY: OS proc 2 maps to socket 0 core 1 thread 0 
        OMP: Info #172: KMP_AFFINITY: OS proc 3 maps to socket 0 core 1 thread 1 
        OMP: Info #172: KMP_AFFINITY: OS proc 4 maps to socket 0 core 2 thread 0 
        OMP: Info #172: KMP_AFFINITY: OS proc 5 maps to socket 0 core 2 thread 1 
        OMP: Info #172: KMP_AFFINITY: OS proc 6 maps to socket 0 core 4 thread 0 
        OMP: Info #172: KMP_AFFINITY: OS proc 7 maps to socket 0 core 4 thread 1 
        OMP: Info #172: KMP_AFFINITY: OS proc 8 maps to socket 0 core 5 thread 0 
        OMP: Info #172: KMP_AFFINITY: OS proc 9 maps to socket 0 core 5 thread 1 
        OMP: Info #172: KMP_AFFINITY: OS proc 10 maps to socket 0 core 6 thread 0 
        OMP: Info #172: KMP_AFFINITY: OS proc 11 maps to socket 0 core 6 thread 1 
        OMP: Info #254: KMP_AFFINITY: pid 32651 tid 32684 thread 1 bound to OS proc set 2
        OMP: Info #254: KMP_AFFINITY: pid 32651 tid 32686 thread 2 bound to OS proc set 4
        OMP: Info #254: KMP_AFFINITY: pid 32651 tid 32687 thread 3 bound to OS proc set 6
        OMP: Info #254: KMP_AFFINITY: pid 32651 tid 32688 thread 4 bound to OS proc set 8
        OMP: Info #254: KMP_AFFINITY: pid 32651 tid 32689 thread 5 bound to OS proc set 10
        OMP: Info #254: KMP_AFFINITY: pid 32651 tid 32690 thread 6 bound to OS proc set 1
        OMP: Info #254: KMP_AFFINITY: pid 32651 tid 32691 thread 7 bound to OS proc set 3
        OMP: Info #254: KMP_AFFINITY: pid 32651 tid 32692 thread 8 bound to OS proc set 5
        OMP: Info #254: KMP_AFFINITY: pid 32651 tid 32694 thread 10 bound to OS proc set 9
        OMP: Info #254: KMP_AFFINITY: pid 32651 tid 32693 thread 9 bound to OS proc set 7
        OMP: Info #254: KMP_AFFINITY: pid 32651 tid 32695 thread 11 bound to OS proc set 11
        OMP: Info #254: KMP_AFFINITY: pid 32651 tid 32696 thread 12 bound to OS proc set 0
        OMP: Info #254: KMP_AFFINITY: pid 32651 tid 32683 thread 13 bound to OS proc set 2
        OMP: Info #254: KMP_AFFINITY: pid 32651 tid 32697 thread 14 bound to OS proc set 4
        OMP: Info #254: KMP_AFFINITY: pid 32651 tid 32698 thread 15 bound to OS proc set 6
        OMP: Info #254: KMP_AFFINITY: pid 32651 tid 32699 thread 16 bound to OS proc set 8
        OMP: Info #254: KMP_AFFINITY: pid 32651 tid 32700 thread 17 bound to OS proc set 10
        OMP: Info #254: KMP_AFFINITY: pid 32651 tid 32701 thread 18 bound to OS proc set 1
        OMP: Info #254: KMP_AFFINITY: pid 32651 tid 32702 thread 19 bound to OS proc set 3
        OMP: Info #254: KMP_AFFINITY: pid 32651 tid 32703 thread 20 bound to OS proc set 5
        OMP: Info #254: KMP_AFFINITY: pid 32651 tid 32704 thread 21 bound to OS proc set 7
        OMP: Info #254: KMP_AFFINITY: pid 32651 tid 32705 thread 22 bound to OS proc set 9
        OMP: Info #254: KMP_AFFINITY: pid 32651 tid 32706 thread 23 bound to OS proc set 11
        OMP: Info #254: KMP_AFFINITY: pid 32651 tid 32707 thread 24 bound to OS proc set 0
        DEEPMD INFO    training data with min nbor dist: 0.7534431439000763
        DEEPMD INFO    training data with max nbor size: [14 26]
        DEEPMD INFO     _____               _____   __  __  _____           _     _  _   
        DEEPMD INFO    |  __ \             |  __ \ |  \/  ||  __ \         | |   (_)| |  
        DEEPMD INFO    | |  | |  ___   ___ | |__) || \  / || |  | | ______ | | __ _ | |_ 
        DEEPMD INFO    | |  | | / _ \ / _ \|  ___/ | |\/| || |  | ||______|| |/ /| || __|
        DEEPMD INFO    | |__| ||  __/|  __/| |     | |  | || |__| |        |   < | || |_ 
        DEEPMD INFO    |_____/  \___| \___||_|     |_|  |_||_____/         |_|\_\|_| \__|
        DEEPMD INFO    Please read and cite:
        DEEPMD INFO    Wang, Zhang, Han and E, Comput.Phys.Comm. 228, 178-184 (2018)
        DEEPMD INFO    installed to:         /home/conda/feedstock_root/build_artifacts/deepmd-kit_1663923506439/work/_skbuild/linux-x86_64-3.10/cmake-install
        DEEPMD INFO    source :              v2.1.5
        DEEPMD INFO    source brach:         HEAD
        DEEPMD INFO    source commit:        6e3d4a62
        DEEPMD INFO    source commit at:     2022-09-23 16:10:28 +0800
        DEEPMD INFO    build float prec:     double
        DEEPMD INFO    build variant:        cpu
        DEEPMD INFO    build with tf inc:    /home/yunai/E/PG/Anaconda/envs/deepmd2.1.5/lib/python3.10/site-packages/tensorflow/include;/home/yunai/E/PG/Anaconda/envs/deepmd2.1.5/include
        DEEPMD INFO    build with tf lib:    
        DEEPMD INFO    ---Summary of the training---------------------------------------
        DEEPMD INFO    running on:           yunai
        DEEPMD INFO    computing device:     cpu:0
        DEEPMD INFO    CUDA_VISIBLE_DEVICES: unset
        DEEPMD INFO    Count of visible GPU: 0
        DEEPMD INFO    num_intra_threads:    0
        DEEPMD INFO    num_inter_threads:    0
        DEEPMD INFO    -----------------------------------------------------------------
        DEEPMD INFO    ---Summary of DataSystem: training     -----------------------------------------------
        DEEPMD INFO    found 2 system(s):
        DEEPMD INFO                                        system  natoms  bch_sz   n_bch   prob  pbc
        DEEPMD INFO                 ./training_data/atomic_system     192       1      80  0.500    F
        DEEPMD INFO                 ./training_data/global_system     192       1      80  0.500    F
        DEEPMD INFO    --------------------------------------------------------------------------------------
        DEEPMD INFO    ---Summary of DataSystem: validation   -----------------------------------------------
        DEEPMD INFO    found 2 system(s):
        DEEPMD INFO                                        system  natoms  bch_sz   n_bch   prob  pbc
        DEEPMD INFO               ./validation_data/atomic_system     192       1      80  0.500    F
        DEEPMD INFO               ./validation_data/global_system     192       1      80  0.500    F
        DEEPMD INFO    --------------------------------------------------------------------------------------
        DEEPMD INFO    training without frame parameter
        DEEPMD INFO    data stating... (this step may take long time)
        DEEPMD INFO    built lr
        DEEPMD INFO    built network
        DEEPMD INFO    built training
        DEEPMD WARNING To get the best performance, it is recommended to adjust the number of threads by setting the environment variables OMP_NUM_THREADS, TF_INTRA_OP_PARALLELISM_THREADS, and TF_INTER_OP_PARALLELISM_THREADS.
        DEEPMD INFO    initialize model from scratch
        DEEPMD INFO    start training at lr 1.00e-02 (== 1.00e-02), decay_step 100, decay_rate 0.501187, final lr will be 1.00e-08
        DEEPMD INFO    batch     100 training time 5.92 s, testing time 0.12 s
        DEEPMD INFO    batch     200 training time 5.61 s, testing time 0.12 s
        DEEPMD INFO    batch     300 training time 5.66 s, testing time 0.12 s
        DEEPMD INFO    batch     400 training time 5.64 s, testing time 0.11 s
        DEEPMD INFO    batch     500 training time 5.64 s, testing time 0.12 s
        DEEPMD INFO    batch     600 training time 5.74 s, testing time 0.12 s
        DEEPMD INFO    batch     700 training time 5.87 s, testing time 0.12 s
        DEEPMD INFO    batch     800 training time 5.79 s, testing time 0.12 s
        DEEPMD INFO    batch     900 training time 5.65 s, testing time 0.12 s
        DEEPMD INFO    batch    1000 training time 5.65 s, testing time 0.11 s
        INFO:tensorflow:/home/yunai/E/SW/DPMD/deepmd-kit-2.1.5/examples/water_tensor/dipole/model.ckpt-1000.data-00000-of-00001
        DEEPMD INFO    /home/yunai/E/SW/DPMD/deepmd-kit-2.1.5/examples/water_tensor/dipole/model.ckpt-1000.data-00000-of-00001
        INFO:tensorflow:2800
        DEEPMD INFO    2800
        INFO:tensorflow:/home/yunai/E/SW/DPMD/deepmd-kit-2.1.5/examples/water_tensor/dipole/model.ckpt-1000.index
        DEEPMD INFO    /home/yunai/E/SW/DPMD/deepmd-kit-2.1.5/examples/water_tensor/dipole/model.ckpt-1000.index
        INFO:tensorflow:2800
        DEEPMD INFO    2800
        INFO:tensorflow:/home/yunai/E/SW/DPMD/deepmd-kit-2.1.5/examples/water_tensor/dipole/model.ckpt-1000.meta
        DEEPMD INFO    /home/yunai/E/SW/DPMD/deepmd-kit-2.1.5/examples/water_tensor/dipole/model.ckpt-1000.meta
        INFO:tensorflow:3800
        DEEPMD INFO    3800
        DEEPMD INFO    saved checkpoint model.ckpt
        DEEPMD INFO    batch    1100 training time 6.03 s, testing time 0.12 s
        DEEPMD INFO    batch    1200 training time 5.63 s, testing time 0.12 s
        DEEPMD INFO    batch    1300 training time 5.60 s, testing time 0.12 s
        DEEPMD INFO    batch    1400 training time 5.87 s, testing time 0.16 s
        DEEPMD INFO    batch    1500 training time 5.82 s, testing time 0.12 s
        DEEPMD INFO    batch    1600 training time 5.80 s, testing time 0.11 s
        DEEPMD INFO    batch    1700 training time 5.61 s, testing time 0.11 s
        DEEPMD INFO    batch    1800 training time 5.59 s, testing time 0.12 s
        DEEPMD INFO    batch    1900 training time 5.59 s, testing time 0.12 s
        DEEPMD INFO    batch    2000 training time 5.59 s, testing time 0.12 s
        INFO:tensorflow:/home/yunai/E/SW/DPMD/deepmd-kit-2.1.5/examples/water_tensor/dipole/model.ckpt-2000.data-00000-of-00001
        DEEPMD INFO    /home/yunai/E/SW/DPMD/deepmd-kit-2.1.5/examples/water_tensor/dipole/model.ckpt-2000.data-00000-of-00001
        INFO:tensorflow:2800
        DEEPMD INFO    2800
        INFO:tensorflow:/home/yunai/E/SW/DPMD/deepmd-kit-2.1.5/examples/water_tensor/dipole/model.ckpt-2000.index
        DEEPMD INFO    /home/yunai/E/SW/DPMD/deepmd-kit-2.1.5/examples/water_tensor/dipole/model.ckpt-2000.index
        INFO:tensorflow:2800
        DEEPMD INFO    2800
        INFO:tensorflow:/home/yunai/E/SW/DPMD/deepmd-kit-2.1.5/examples/water_tensor/dipole/model.ckpt-2000.meta
        DEEPMD INFO    /home/yunai/E/SW/DPMD/deepmd-kit-2.1.5/examples/water_tensor/dipole/model.ckpt-2000.meta
        INFO:tensorflow:3800
        DEEPMD INFO    3800
        DEEPMD INFO    saved checkpoint model.ckpt
        DEEPMD INFO    finished training
        DEEPMD INFO    wall time: 118.161 s

### The freeze and compress log
        WARNING:tensorflow:From /home/yunai/E/PG/Anaconda/envs/deepmd2.1.5/lib/python3.10/site-packages/tensorflow/python/compat/v2_compat.py:107: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.
        Instructions for updating:
        non-resource variables are not supported in the long term
        WARNING:root:To get the best performance, it is recommended to adjust the number of threads by setting the environment variables OMP_NUM_THREADS, TF_INTRA_OP_PARALLELISM_THREADS, and TF_INTER_OP_PARALLELISM_THREADS.
        WARNING:root:Environment variable KMP_BLOCKTIME is empty. Use the default value 0
        WARNING:root:Environment variable KMP_AFFINITY is empty. Use the default value granularity=fine,verbose,compact,1,0
        /home/yunai/E/PG/Anaconda/envs/deepmd2.1.5/lib/python3.10/importlib/__init__.py:169: UserWarning: The NumPy module was reloaded (imported a second time). This can in some cases result in small but subtle issues and is discouraged.
          _bootstrap._exec(spec, module)
        INFO:tensorflow:Restoring parameters from ./model.ckpt-2000
        DEEPMD INFO    Restoring parameters from ./model.ckpt-2000
        DEEPMD INFO    The following nodes will be frozen: ['model_type', 'descrpt_attr/rcut', 'descrpt_attr/ntypes', 'model_attr/tmap', 'model_attr/model_type', 'model_attr/model_version', 'train_attr/min_nbor_dist', 'train_attr/training_script', 'o_dipole', 'o_global_dipole', 'o_force', 'o_virial', 'o_atom_virial', 'o_rmat', 'o_rmat_deriv', 'o_nlist', 'o_rij', 'descrpt_attr/sel', 'descrpt_attr/ndescrpt', 'model_attr/sel_type', 'model_attr/output_dim']
        WARNING:tensorflow:From /home/yunai/E/PG/Anaconda/envs/deepmd2.1.5/lib/python3.10/site-packages/deepmd/entrypoints/freeze.py:246: convert_variables_to_constants (from tensorflow.python.framework.graph_util_impl) is deprecated and will be removed in a future version.
        Instructions for updating:
        Use `tf.compat.v1.graph_util.convert_variables_to_constants`
        DEEPMD WARNING From /home/yunai/E/PG/Anaconda/envs/deepmd2.1.5/lib/python3.10/site-packages/deepmd/entrypoints/freeze.py:246: convert_variables_to_constants (from tensorflow.python.framework.graph_util_impl) is deprecated and will be removed in a future version.
        Instructions for updating:
        Use `tf.compat.v1.graph_util.convert_variables_to_constants`
        WARNING:tensorflow:From /home/yunai/E/PG/Anaconda/envs/deepmd2.1.5/lib/python3.10/site-packages/tensorflow/python/framework/convert_to_constants.py:925: extract_sub_graph (from tensorflow.python.framework.graph_util_impl) is deprecated and will be removed in a future version.
        Instructions for updating:
        Use `tf.compat.v1.graph_util.extract_sub_graph`
        DEEPMD WARNING From /home/yunai/E/PG/Anaconda/envs/deepmd2.1.5/lib/python3.10/site-packages/tensorflow/python/framework/convert_to_constants.py:925: extract_sub_graph (from tensorflow.python.framework.graph_util_impl) is deprecated and will be removed in a future version.
        Instructions for updating:
        Use `tf.compat.v1.graph_util.extract_sub_graph`
        DEEPMD INFO    2485 ops in the final graph.
        WARNING:tensorflow:From /home/yunai/E/PG/Anaconda/envs/deepmd2.1.5/lib/python3.10/site-packages/tensorflow/python/compat/v2_compat.py:107: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.
        Instructions for updating:
        non-resource variables are not supported in the long term
        WARNING:root:To get the best performance, it is recommended to adjust the number of threads by setting the environment variables OMP_NUM_THREADS, TF_INTRA_OP_PARALLELISM_THREADS, and TF_INTER_OP_PARALLELISM_THREADS.
        WARNING:root:Environment variable KMP_BLOCKTIME is empty. Use the default value 0
        WARNING:root:Environment variable KMP_AFFINITY is empty. Use the default value granularity=fine,verbose,compact,1,0
        /home/yunai/E/PG/Anaconda/envs/deepmd2.1.5/lib/python3.10/importlib/__init__.py:169: UserWarning: The NumPy module was reloaded (imported a second time). This can in some cases result in small but subtle issues and is discouraged.
          _bootstrap._exec(spec, module)
        DEEPMD INFO    
        
        
        DEEPMD INFO    stage 1: compress the model
        DEEPMD INFO     _____               _____   __  __  _____           _     _  _   
        DEEPMD INFO    |  __ \             |  __ \ |  \/  ||  __ \         | |   (_)| |  
        DEEPMD INFO    | |  | |  ___   ___ | |__) || \  / || |  | | ______ | | __ _ | |_ 
        DEEPMD INFO    | |  | | / _ \ / _ \|  ___/ | |\/| || |  | ||______|| |/ /| || __|
        DEEPMD INFO    | |__| ||  __/|  __/| |     | |  | || |__| |        |   < | || |_ 
        DEEPMD INFO    |_____/  \___| \___||_|     |_|  |_||_____/         |_|\_\|_| \__|
        DEEPMD INFO    Please read and cite:
        DEEPMD INFO    Wang, Zhang, Han and E, Comput.Phys.Comm. 228, 178-184 (2018)
        DEEPMD INFO    installed to:         /home/conda/feedstock_root/build_artifacts/deepmd-kit_1663923506439/work/_skbuild/linux-x86_64-3.10/cmake-install
        DEEPMD INFO    source :              v2.1.5
        DEEPMD INFO    source brach:         HEAD
        DEEPMD INFO    source commit:        6e3d4a62
        DEEPMD INFO    source commit at:     2022-09-23 16:10:28 +0800
        DEEPMD INFO    build float prec:     double
        DEEPMD INFO    build variant:        cpu
        DEEPMD INFO    build with tf inc:    /home/yunai/E/PG/Anaconda/envs/deepmd2.1.5/lib/python3.10/site-packages/tensorflow/include;/home/yunai/E/PG/Anaconda/envs/deepmd2.1.5/include
        DEEPMD INFO    build with tf lib:    
        DEEPMD INFO    ---Summary of the training---------------------------------------
        DEEPMD INFO    running on:           yunai
        DEEPMD INFO    computing device:     cpu:0
        DEEPMD INFO    CUDA_VISIBLE_DEVICES: unset
        DEEPMD INFO    Count of visible GPU: 0
        DEEPMD INFO    num_intra_threads:    0
        DEEPMD INFO    num_inter_threads:    0
        DEEPMD INFO    -----------------------------------------------------------------
        DEEPMD INFO    training without frame parameter
        DEEPMD INFO    training data with lower boundary: [-0.31993986 -0.32837175]
        DEEPMD INFO    training data with upper boundary: [8.18718375 9.02269379]
        OMP: Info #155: KMP_AFFINITY: Initial OS proc set respected: 0-11
        OMP: Info #216: KMP_AFFINITY: decoding x2APIC ids.
        OMP: Info #157: KMP_AFFINITY: 12 available OS procs
        OMP: Info #158: KMP_AFFINITY: Uniform topology
        OMP: Info #287: KMP_AFFINITY: topology layer "LL cache" is equivalent to "socket".
        OMP: Info #192: KMP_AFFINITY: 1 socket x 6 cores/socket x 2 threads/core (6 total cores)
        OMP: Info #218: KMP_AFFINITY: OS proc to physical thread map:
        OMP: Info #172: KMP_AFFINITY: OS proc 0 maps to socket 0 core 0 thread 0 
        OMP: Info #172: KMP_AFFINITY: OS proc 1 maps to socket 0 core 0 thread 1 
        OMP: Info #172: KMP_AFFINITY: OS proc 2 maps to socket 0 core 1 thread 0 
        OMP: Info #172: KMP_AFFINITY: OS proc 3 maps to socket 0 core 1 thread 1 
        OMP: Info #172: KMP_AFFINITY: OS proc 4 maps to socket 0 core 2 thread 0 
        OMP: Info #172: KMP_AFFINITY: OS proc 5 maps to socket 0 core 2 thread 1 
        OMP: Info #172: KMP_AFFINITY: OS proc 6 maps to socket 0 core 4 thread 0 
        OMP: Info #172: KMP_AFFINITY: OS proc 7 maps to socket 0 core 4 thread 1 
        OMP: Info #172: KMP_AFFINITY: OS proc 8 maps to socket 0 core 5 thread 0 
        OMP: Info #172: KMP_AFFINITY: OS proc 9 maps to socket 0 core 5 thread 1 
        OMP: Info #172: KMP_AFFINITY: OS proc 10 maps to socket 0 core 6 thread 0 
        OMP: Info #172: KMP_AFFINITY: OS proc 11 maps to socket 0 core 6 thread 1 
        OMP: Info #254: KMP_AFFINITY: pid 33346 tid 33346 thread 0 bound to OS proc set 0
        OMP: Info #254: KMP_AFFINITY: pid 33346 tid 33393 thread 1 bound to OS proc set 2
        OMP: Info #254: KMP_AFFINITY: pid 33346 tid 33394 thread 2 bound to OS proc set 4
        OMP: Info #254: KMP_AFFINITY: pid 33346 tid 33395 thread 3 bound to OS proc set 6
        OMP: Info #254: KMP_AFFINITY: pid 33346 tid 33396 thread 4 bound to OS proc set 8
        OMP: Info #254: KMP_AFFINITY: pid 33346 tid 33397 thread 5 bound to OS proc set 10
        DEEPMD INFO    built lr
        DEEPMD INFO    built network
        DEEPMD INFO    built training
        DEEPMD WARNING To get the best performance, it is recommended to adjust the number of threads by setting the environment variables OMP_NUM_THREADS, TF_INTRA_OP_PARALLELISM_THREADS, and TF_INTER_OP_PARALLELISM_THREADS.
        DEEPMD INFO    initialize model from scratch
        INFO:tensorflow:/home/yunai/E/SW/DPMD/deepmd-kit-2.1.5/examples/water_tensor/dipole/model-compression/model.ckpt.data-00000-of-00001
        DEEPMD INFO    /home/yunai/E/SW/DPMD/deepmd-kit-2.1.5/examples/water_tensor/dipole/model-compression/model.ckpt.data-00000-of-00001
        INFO:tensorflow:2200
        DEEPMD INFO    2200
        INFO:tensorflow:/home/yunai/E/SW/DPMD/deepmd-kit-2.1.5/examples/water_tensor/dipole/model-compression/model.ckpt.index
        DEEPMD INFO    /home/yunai/E/SW/DPMD/deepmd-kit-2.1.5/examples/water_tensor/dipole/model-compression/model.ckpt.index
        INFO:tensorflow:2200
        DEEPMD INFO    2200
        INFO:tensorflow:/home/yunai/E/SW/DPMD/deepmd-kit-2.1.5/examples/water_tensor/dipole/model-compression/model.ckpt.meta
        DEEPMD INFO    /home/yunai/E/SW/DPMD/deepmd-kit-2.1.5/examples/water_tensor/dipole/model-compression/model.ckpt.meta
        INFO:tensorflow:30900
        DEEPMD INFO    30900
        DEEPMD INFO    finished compressing
        DEEPMD INFO    
        
        
        DEEPMD INFO    stage 2: freeze the model
        INFO:tensorflow:Restoring parameters from model-compression/model.ckpt
        DEEPMD INFO    Restoring parameters from model-compression/model.ckpt
        DEEPMD INFO    The following nodes will be frozen: ['model_type', 'descrpt_attr/rcut', 'descrpt_attr/ntypes', 'model_attr/tmap', 'model_attr/model_type', 'model_attr/model_version', 'train_attr/min_nbor_dist', 'train_attr/training_script', 'o_dipole', 'o_global_dipole', 'o_force', 'o_virial', 'o_atom_virial', 'o_rmat', 'o_rmat_deriv', 'o_nlist', 'o_rij', 'descrpt_attr/sel', 'descrpt_attr/ndescrpt', 'model_attr/sel_type', 'model_attr/output_dim']
        WARNING:tensorflow:From /home/yunai/E/PG/Anaconda/envs/deepmd2.1.5/lib/python3.10/site-packages/deepmd/entrypoints/freeze.py:246: convert_variables_to_constants (from tensorflow.python.framework.graph_util_impl) is deprecated and will be removed in a future version.
        Instructions for updating:
        Use `tf.compat.v1.graph_util.convert_variables_to_constants`
        DEEPMD WARNING From /home/yunai/E/PG/Anaconda/envs/deepmd2.1.5/lib/python3.10/site-packages/deepmd/entrypoints/freeze.py:246: convert_variables_to_constants (from tensorflow.python.framework.graph_util_impl) is deprecated and will be removed in a future version.
        Instructions for updating:
        Use `tf.compat.v1.graph_util.convert_variables_to_constants`
        WARNING:tensorflow:From /home/yunai/E/PG/Anaconda/envs/deepmd2.1.5/lib/python3.10/site-packages/tensorflow/python/framework/convert_to_constants.py:925: extract_sub_graph (from tensorflow.python.framework.graph_util_impl) is deprecated and will be removed in a future version.
        Instructions for updating:
        Use `tf.compat.v1.graph_util.extract_sub_graph`
        DEEPMD WARNING From /home/yunai/E/PG/Anaconda/envs/deepmd2.1.5/lib/python3.10/site-packages/tensorflow/python/framework/convert_to_constants.py:925: extract_sub_graph (from tensorflow.python.framework.graph_util_impl) is deprecated and will be removed in a future version.
        Instructions for updating:
        Use `tf.compat.v1.graph_util.extract_sub_graph`
        DEEPMD INFO    1657 ops in the final graph.

**Comment by njzjz** [comment, marked as answer]
You should use a potential energy model for pair_style deepmd.

**Reply by yunai2384** [reply]
I got it, thank you very much for your tips !
